nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp

```diff
@@ -436,21 +492,11 @@ VBufStorage_fieldNode_t* GeckoVBufBackend_t::fillVBuf(
 	}
 
 	//Get role -- IAccessible2 role
-	long role=0;
-	BSTR roleString=NULL;
-	if(pacc->role(&role)!=S_OK)
-		role=IA2_ROLE_UNKNOWN;
-	VARIANT varRole;
-	VariantInit(&varRole);
-	if(role==0) {
-		if(pacc->get_accRole(varChild,&varRole)!=S_OK) {
-			LOG_DEBUG(L"accRole failed");
-		}
-		if(varRole.vt==VT_I4)
-			role=varRole.lVal;
-		else if(varRole.vt==VT_BSTR)
-			roleString=varRole.bstrVal;
-	}
+	long role = 0;
+	CComBSTR roleString;
+	CComPtr<IAccessible2> smartPacc = CComQIPtr<IAccessible2>(pacc);
+	CComVariant smartVariant = CComVariant(&varChild);
+	std::tie(role, roleString) = getRoleLongRoleString(smartPacc, smartVariant);

```

Comment:
Probably a good idea to limit the scope here. The scope of `smartPacc` and `smartVariant` defines when they will do their clean up. You can add arbitrary `{}` characters to limit scope, though in this case you might consider just passing in the two parameters directly.

 ```suggestion
	std::tie(role, roleString) = getRoleLongRoleString(CComQIPtr<IAccessible2>(pacc), CComVariant(&varChild));

```

Reply:
`smartPacc` is used further down in the same scope by `fillVBufAriaDetails`. I can make this change for `smartVariant`